src/ansys/fluent/core/workflow.py

```diff
@@ -3,14 +3,51 @@
 from __future__ import annotations
 
 import logging
+import re
 import threading
-from typing import Any, Iterator, List, Optional, Tuple, Union
+from typing import Any, Iterable, Iterator, List, Optional, Tuple, Union
 import warnings
 
 import ansys.fluent.core as pyfluent
 from ansys.fluent.core.data_model_cache import DataModelCache
 from ansys.fluent.core.services.datamodel_se import PyCallableStateObject, PyMenuGeneric
 
+
+def camel_to_snake_case(camel_case_str: str) -> str:
+    """Convert camel case input string to snake case output string."""
+    try:
+        return camel_to_snake_case.cache[camel_case_str]
+    except KeyError:
+        camel_to_snake_case.cache[camel_case_str] = re.sub(
+            "((?<=[a-z0-9])[A-Z]|(?!^)[A-Z](?=[a-z]))", r"_\1", camel_case_str
+        ).lower()
+        return camel_to_snake_case.cache[camel_case_str]
+
+
+camel_to_snake_case.cache = {}
+
+
+def snake_to_camel_case(snake_case_str: str, camel_case_strs: Iterable):
+    """Populate the snake-case attribute map and return camel case of the passed
+    attribute."""
+    try:
+        return snake_to_camel_case.cache[snake_case_str]
+    except KeyError:
+        if snake_case_str.islower():
+            for camel_case_str in camel_case_strs:
+                _snake_case_str = camel_to_snake_case(str(camel_case_str))
+                if _snake_case_str not in snake_to_camel_case.cache:
+                    snake_to_camel_case.cache[_snake_case_str] = str(camel_case_str)
+
+        try:
+            return snake_to_camel_case.cache[snake_case_str]
+        except KeyError:
+            return
```

Comment:
If the result must be one of `camel_case_strs`, prefer to throw exception. That will catch programming error elsewhere.